api/app/sfms/daily_ffmc_processor.py

```diff
@@ -0,0 +1,69 @@
+import os
+import logging
+import tempfile
+from typing import Callable, List, Iterator, cast
+from datetime import datetime, timedelta
+from app.auto_spatial_advisory.run_type import RunType
+from app.geospatial.wps_dataset import WPSDataset
+from app.sfms.fwi_processor import calculate_ffmc
+from app.sfms.raster_addresser import RasterKeyAddresser, WeatherParameter
+from app.utils.geospatial import GDALResamplingMethod
+from app.utils.s3_client import S3Client
+
+logger = logging.getLogger(__name__)
+
+DAILY_FFMC_DAYS = 2
+
+MultiDatasetContext = Callable[[List[str]], Iterator[List["WPSDataset"]]]
+
+
+class DailyFFMCProcessor:
+    """
+    Class for calculating/generating forecasted daily FFMC rasters for a date range
+    """
+
+    def __init__(self, start_datetime: datetime, addresser: RasterKeyAddresser):
+        self.start_datetime = start_datetime
+        self.addresser = addresser
+
+    async def process_daily_ffmc(self, s3_client: S3Client, input_dataset_context: MultiDatasetContext):
+        for day in range(DAILY_FFMC_DAYS):
+            current_day = self.start_datetime + timedelta(days=day)
+            yesterday = current_day - timedelta(days=1)
+            yesterday_ffmc_key = self.addresser.get_daily_ffmc(yesterday, RunType.ACTUAL)
+            if await s3_client.all_objects_exist(yesterday_ffmc_key) == False:
+                yesterday_ffmc_key = self.addresser.get_calculated_daily_ffmc(yesterday)
+
+            if await s3_client.all_objects_exist(yesterday_ffmc_key) == False:
+                logging.warning(f"No ffmc objects found for key: {yesterday_ffmc_key}")
+                return
+
+            temp_forecast_key = self.addresser.get_daily_model_data_key(current_day, RunType.FORECAST, WeatherParameter.TEMP)
```

Comment:
I think if we want to use weather_model data we need to keep track of a prediction_hour because we get model data for so many different hours, but maybe there's a better way to do it

Reply:
Opted to use all the daily SFMS model data since that's where we get the previous FFMC raster. Not sure if that's correct though. I guess that conflicts with our BUI calculations but the seed FFMC raster will either way.

Reply:
Gotcha, I could be wrong but I think the intention was to step forward in these calculations with weather model data (RDPS), since the weather data coming from SFMS is the coarsely interpolated forecasts from BC weather stations